src/backend/commands/matview.c

```diff
@@ -1302,13 +1316,33 @@ IVM_immediate_maintenance(PG_FUNCTION_ARGS)
 		/* loop for self-join */
 		foreach(lc2, table->rte_indexes)
 		{
-			int index = lfirst_int(lc2);
+			List *index_paths = lfirst(lc2);
```

Comment:
Maybe, index_paths is misleading because this is a path to a certain RTE, so I think index_path is better.

Reply:
OK, I will fix it.